File: Python/38. Count and Say.py
class Solution:
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        ans = "1"
        while n > 1:
            cnt, tmp = 0, ""
            p = ans[0]
            for c in ans:
                if c == p:
                    cnt += 1
                else:
                    tmp += str(cnt) + p
                    p = c
                    cnt = 1
            tmp += str(cnt) + p
            ans = tmp
            n -= 1
        return ans

File: C++/38. Count and Say.cpp
class Solution {
public:
    string countAndSay(int n) {
        string ans = "1";
        while(--n)
        {
            string tmp = "";
            int cnt = 0;
            char p = ans[0];
            for(auto c: ans)
            {
                if(p == c)
                    cnt += 1;
                else
                {
                    tmp += to_string(cnt) + p;
                    p = c;
                    cnt = 1;
                }
            }
            tmp += to_string(cnt) + p;
            ans = tmp;
        }
        return ans;
    }
};